velox/dwio/common/TypeUtils.cpp

```diff
@@ -43,7 +43,12 @@ std::shared_ptr<const TypeWithId> visit(
     const std::shared_ptr<const TypeWithId>& typeWithId,
     const std::function<bool(size_t)>& selector) {
   if (typeWithId->type()->isPrimitiveType()) {
-    return typeWithId;
+    return std::make_shared<TypeWithId>(
+        typeWithId->type(),
+        std::vector<std::shared_ptr<const TypeWithId>>(),
+        typeWithId->id(),
+        typeWithId->maxId(),
+        typeWithId->column());
   }
   if (typeWithId->type()->isRow()) {
     std::vector<std::string> names;
```

Comment:
This comment is for the next line. I know this is not related, but I wonder if we can change `typesWithId` to `selectedChildren`.  The `typesWithId` is very easy to be confused with the input parameter `typeWithId`